connector/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchSourceSuite.scala

```diff
@@ -666,10 +669,7 @@ abstract class KafkaMicroBatchSourceSuiteBase extends KafkaSourceSuiteBase {
         testUtils.sendMessages(topic2, Array("6"))
       },
       StartStream(),
-      ExpectFailure[IllegalStateException](e => {
-        // The offset of `topic2` should be changed from 2 to 1
-        assert(e.getMessage.contains("was changed from 2 to 1"))
-      })
+      expectOffsetChange()
```

Comment:
I have to change this because there is different behavior of streaming V1 and V2 though the Kafka connector throws an illegal state exception:
1. https://github.com/apache/spark/blob/b11252806d49ed7915744147216db41a747e7d4d/connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchStream.scala#L313
2. https://github.com/apache/spark/blob/b4c019627b676edf850c00bb070377896b66fad2/connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSource.scala#L346

Reply:
@cloud-fan @srielau Do you agree that we wrap `IllegalStateException` by `INTERNAL_ERROR` coming from connectors (like Kafka connector). Precisely speaking, this is not Spark's internal error, and user's should reports bugs to the connector devs.

Reply:
well, if we build a product based on Spark, this is still an internal error of the product.